Libraries/LibWeb/WebAssembly/WebAssembly.cpp

```diff
@@ -851,4 +882,126 @@ GC::Ref<WebIDL::Promise> compile_potential_webassembly_response(JS::VM& vm, GC::
+            /* b. Perform CreateNonEnumerableDataPropertyOrThrow(O, "message", msg). */                                        \
+            error->create_non_enumerable_data_property_or_throw(vm.names.message, JS::PrimitiveString::create(vm, move(msg))); \
+        }                                                                                                                      \
+                                                                                                                               \
+        /* 4. Perform ? InstallErrorCause(O, options). */                                                                      \
+        TRY(error->install_error_cause(options));                                                                              \
+                                                                                                                               \
+        /* 5. Return O. */                                                                                                     \
+        return error;                                                                                                          \
+    }
+
+DEFINE_HACKY_NATIVE_ERROR_CONSTRUCTOR(CompileError, compile_error, CompileErrorPrototype, CompileErrorConstructor)
+DEFINE_HACKY_NATIVE_ERROR_CONSTRUCTOR(LinkError, link_error, LinkErrorPrototype, LinkErrorConstructor)
+DEFINE_HACKY_NATIVE_ERROR_CONSTRUCTOR(RuntimeError, runtime_error, RuntimeErrorPrototype, RuntimeErrorConstructor)
+
+#undef DEFINE_HACKY_NATIVE_ERROR_CONSTRUCTOR
+
+#define DEFINE_HACKY_NATIVE_ERROR_PROTOTYPE(ClassName, snake_name, PrototypeName, ConstructorName)         \
+    GC_DEFINE_ALLOCATOR(PrototypeName);                                                                    \
+                                                                                                           \
+    PrototypeName::PrototypeName(JS::Realm& realm)                                                         \
+        : PrototypeObject(realm.intrinsics().error_prototype())                                            \
+    {                                                                                                      \
+    }                                                                                                      \
+                                                                                                           \
+    GC::Ptr<PrototypeName> g_##snake_name##_prototype_instance { nullptr };                                \
+    PrototypeName& PrototypeName::hacky_the(JS::Realm& realm)                                              \
+    {                                                                                                      \
+        if (!g_##snake_name##_prototype_instance)                                                          \
+            g_##snake_name##_prototype_instance = realm.create<PrototypeName>(realm);                      \
```

Comment:
ehm. what's this then? Why are we creating a global instance of a GC object that's associated with a realm? This will keep the realm alive forever. 

Stash this on the Web Intrinsics or something.

Reply:
Also calling all of this "hacky" is a bit aggresssive. Can we go with DEFINE_WASM_ERROR or something? 😅

Reply:
Sure, I'll put it on the web intrinsic.
🤔 would putting it there change anything re keeping it alive forever?

re hacky: it _is_ hacky though? just copying this out of LibJS is hacky to me, not to mention the prototype mess.

Reply:
>  🤔 would putting it there change anything re keeping it alive forever?

Yes, because the web intrinsics are GC-owned by the realm. Realms are not eternal. They are (assuming no root-leaks) scoped to the lifetime of the window that they represent. We can have multiple WindowGlobalObjects in the same process thanks to iframes. And in the future, when we cram more windows into the 'similar origin window agent'